src/core/CorePool.ts

```diff
@@ -113,10 +133,22 @@ export class CorePool {
     tickUpper: number,
     amount: JSBI
   ): { amount0: JSBI; amount1: JSBI } {
-    // TODO
+    let positionStep = this.modifyPosition(owner, tickLower, tickUpper, amount);
+
+    let amount0 = JSBI.bitwiseNot(positionStep.amount0);
+    let amount1 = JSBI.bitwiseNot(positionStep.amount1);
+
+    if (JSBI.greaterThan(amount0, ZERO) || JSBI.greaterThan(amount1, ZERO)) {
+      let _position = this.getPosition(owner, tickLower, tickUpper);
+
+      let _newTokensOwed0 = JSBI.add(_position.tokensOwed0, amount0);
+      let _newTokensOwed1 = JSBI.add(_position.tokensOwed1, amount1);
+      // TODO, use new TokenOwed values set up new position and update it.
```

Comment:
I see your point and let's finish this after changes on pr #31

Reply:
Ok!